include/fastdds/rtps/writer/ReaderProxy.h

```diff
@@ -145,75 +152,27 @@ class ReaderProxy
      * @return true if at least one change has been marked as REQUESTED, false otherwise.
      */
     bool requested_changes_set(
-            const SequenceNumberSet_t& seq_num_set);
+            const SequenceNumberSet_t& seq_num_set,
+            RTPSGapBuilder& gap_builder);
 
     /**
      * Performs processing of preemptive acknack
      * @return true if a heartbeat should be sent, false otherwise.
      */
-    bool process_initial_acknack();
-
-    /**
-     * Applies the given function object to every unsent change.
-     * @param max_seq Maximum sequence number to be considered without including it.
-     * @param f Function to apply.
-     *          Will receive a SequenceNumber_t and a ChangeForReader_t*.
-     *          The second argument may be nullptr for irrelevant changes.
-     */
-    template <class BinaryFunction>
-    void for_each_unsent_change(
-            const SequenceNumber_t& max_seq,
-            BinaryFunction f) const
-    {
-        if (!changes_for_reader_.empty())
-        {
-            SequenceNumber_t current_seq = changes_low_mark_ + 1;
-            ChangeConstIterator it = changes_for_reader_.begin();
-            while (it != changes_for_reader_.end())
-            {
-                // Holes before this change are informed as irrelevant.
-                SequenceNumber_t change_seq = it->getSequenceNumber();
-                for (; current_seq < change_seq; ++current_seq)
-                {
-                    f(current_seq, nullptr);
-                }
-
-                // We then inform of this change if it is unsent, and go to the next one.
-                if (it->getStatus() == UNSENT)
-                {
-                    f(current_seq, &(*it));
-                }
-                ++current_seq;
-                ++it;
-            }
-
-            // After the last change has been checked, there may be a hole at the end.
-            for (; current_seq < max_seq; ++current_seq)
-            {
-                f(current_seq, nullptr);
-            }
-        }
-        else
-        {
-            // This may be entered if all changes where removed before being acknowledged.
-            for (SequenceNumber_t seq = changes_low_mark_ + 1; seq < max_seq; ++seq)
-            {
-                f(seq, nullptr);
-            }
-        }
-    }
+    bool process_initial_acknack(
+            std::function<void(ChangeForReader_t& change)> func);
```

Comment:
This parameter is not documented

Reply:
Passing in an `std::function` by value produces dynamic allocations. You could pass a const reference, or make this a template method.

Reply:
Done